packaging_automation/package_cloud_statistics_collector.py

```diff
@@ -81,48 +106,105 @@ def fetch_and_save_package_cloud_stats(db_params: DbParams, package_cloud_api_to
+                                                 epoch=package_info['epoch'],
+                                                 download_date=download_date,
+                                                 download_count=download_count,
+                                                 detail_url=package_info['downloads_detail_url'])
+
+            session.add(pc_stats)
+
+
+def fetch_and_save_package_download_details(package_info, package_cloud_api_token: str,
+                                            session,
+                                            repo_name: PackageCloudRepos):
+    print(f"Download Detail Query for {package_info['filename']}: {package_info['downloads_detail_url']}")
+    page_number = 1
+    record_count = DETAIL_PAGE_RECORD_COUNT
+    while record_count == DETAIL_PAGE_RECORD_COUNT:
         request_result = stat_get_request(
-            package_historic_statistics_request_address(package_cloud_api_token, package_info['downloads_series_url']),
+            package_statistics_detail_request_address(package_cloud_api_token, package_info['downloads_detail_url'],
+                                                      DETAIL_PAGE_RECORD_COUNT, page_number),
             RequestType.package_cloud_detail_query, session)
+        page_number = page_number + 1
         if request_result.status_code != HTTPStatus.OK:
-            continue
-        download_stats = json.loads(request_result.content)
-        for stat_date in download_stats['value']:
-            download_date = datetime.strptime(stat_date, PC_DOWNLOAD_DATE_FORMAT).date()
-            download_count = int(download_stats['value'][stat_date])
-            if (download_date != date.today() and not is_ignored_package(package_info['name']) and
-                    not stat_records_exists(download_date, package_info['filename'], session) and
-                    is_download_count_eligible_for_save(download_count, save_records_with_download_count_zero)):
-                pc_stats = PackageCloudDownloadStats(fetch_date=datetime.now(), repo=repo_name,
-                                                     package_full_name=package_info['filename'],
-                                                     package_name=package_info['name'],
-                                                     package_version=package_info['version'],
-                                                     package_release=package_info['release'],
-                                                     package_type=package_info['type'],
-                                                     download_date=download_date,
-                                                     download_count=download_count)
-                session.add(pc_stats)
-
-
-def package_historic_statistics_request_address(package_cloud_api_token: str, detail_query_uri: str):
+            print(
+                f"Error while calling detail query for package {package_info['filename']}. Error Code: {request_result.status_code}")
```

Comment:
Similarly we error here but still commit later

Reply:
Fixrd threw error